### PR TITLE
fix: vars lookup in NWAS HA role include

### PR DESCRIPTION
**Pull request by ja9fuchs**
This solves a problem with the `vars[]` construct, which did not resolve nested variables.

Previously the nested variable names were passed as `{{ ... }}` **string** values in the SAP HA role cluster resource setup.

`- agent: ocf:heartbeat:Filesystem`
`  id: rsc_fs_DEF_ASCS02`
`  instance_attrs:`
`  - attrs:`
`    - name: device`
`      value: '{{ nfs_mount_point | default('''') }}/usr/sap/DEF/ASCS02'`
`    - name: directory`
`      value: /usr/sap/DEF/ASCS02`
`    - name: fstype`
`      value: '{{ nfs_mount_point_type | default('''') }}'`
`    - name: options`
`      value: '{{ nfs_mount_point_opts | default('''') }}'`
`    - name: force_unmount`
`      value: safe`

Now it resolves correctly:

`- agent: ocf:heartbeat:Filesystem`
`  id: rsc_fs_DEF_ASCS02`
`  instance_attrs:`
`  - attrs:`
`    - name: device`
`      value: <...my-EFS-ID...>.amazonaws.com:/usr/sap/DEF/ASCS02`
`    - name: directory`
`      value: /usr/sap/DEF/ASCS02`
`    - name: fstype`
`      value: nfs`
`    - name: options`
`      value: defaults`
`    - name: force_unmount`
`      value: safe`

### CODE REVIEW

File: deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml [hunk 1]
```diff
@@ -611,7 +611,7 @@
 
         sap_ha_pacemaker_cluster_vip_resource_group_name: vipnwasascs
 
-        sap_ha_pacemaker_cluster_storage_definition: "{{ vars['sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary'][sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
+        sap_ha_pacemaker_cluster_storage_definition: "{{ lookup('ansible.builtin.vars', 'sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary')[sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
```
Comment (Wabri) [inline deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml:614]: did you try it? I think you should remove the ansible.builtin from it:
```suggestion
        sap_ha_pacemaker_cluster_storage_definition: "{{ lookup('vars', 'sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary')[sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
```

File: deploy_scenarios/sap_s4hana_distributed_ha_maintplan/ansible_playbook.yml [hunk 2]
```diff
@@ -645,7 +645,7 @@
 
         sap_ha_pacemaker_cluster_vip_resource_group_name: vipnwasascs
 
-        sap_ha_pacemaker_cluster_storage_definition: "{{ vars['sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary'][sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
+        sap_ha_pacemaker_cluster_storage_definition: "{{ lookup('ansible.builtin.vars', 'sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary')[sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
```
Comment (Wabri) [inline deploy_scenarios/sap_s4hana_distributed_ha_maintplan/ansible_playbook.yml:648]: Same as above:
```suggestion
        sap_ha_pacemaker_cluster_storage_definition: "{{ lookup('vars', 'sap_vm_provision_' + sap_vm_provision_iac_platform + '_host_specifications_dictionary')[sap_vm_provision_host_specification_plan][ansible_hostname].storage_definition | list }}"
```

**Review by Wabri — CHANGES_REQUESTED** [review]

File: deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml (hunk 1 above)
Comment (Wabri) [reply deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml:614]: I've check it and it have the same results!

File: deploy_scenarios/sap_s4hana_distributed_ha_maintplan/ansible_playbook.yml (hunk 2 above)
Comment (Wabri) [reply deploy_scenarios/sap_s4hana_distributed_ha_maintplan/ansible_playbook.yml:648]: I've check it and it have the same results!

**Review by Wabri — APPROVED** [review]

**Review by marcelmamula — APPROVED** [review]
LGTM

File: deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml (hunk 1 above)
Comment (ja9fuchs) [reply deploy_scenarios/sap_s4hana_distributed_ha/ansible_playbook.yml:614]: It is recommended to use the FQCN.
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/vars_lookup.html

**Pull request merged**